addons/controllers/tanzuclusterbootstrap_controller.go

```diff
@@ -584,3 +592,39 @@ func (r *TanzuClusterBootstrapReconciler) watchProvider(providerRef *corev1.Type
 		},
 	)
 }
+
+func (r *TanzuClusterBootstrapReconciler) reconcileClusterProxySettings(cluster *clusterapiv1beta1.Cluster,
+	tcb *runtanzuv1alpha3.TanzuClusterBootstrap,
+	log logr.Logger) error {
+
+	// use patchHelper to auto detect if there is diff in Cluster CR when performing update
+	patchHelper, err := clusterapipatchutil.NewHelper(cluster, r.Client)
+	if err != nil {
+		return err
+	}
+
+	HTTPProxy := ""
+	HTTPSProxy := ""
+	NoProxy := ""
+	if tcb.Spec.Proxy != nil {
```

Comment:
If tcb.Spec.Proxy be nil non of these changes are needed, could you add an if block at the very beginning of the function and immediately return on the following?
```
if tcb.Spec.Proxy != nil {
       return nil
}
```

Reply:
`tcb.Spec.Proxy == nil` means those proxy configs in annotations would be empty. We still want to try patch it, i.e. cluster annotations had some pre-existing proxy annotations and we want to rewrite to empty (User changes proxy to empty).

Reply:
Ok, I see. Thanks for the reply.